cibuildwheel/environment.py

```diff
@@ -74,8 +70,14 @@ def as_dictionary(self, prev_environment):
 
         return environment
 
-    def as_shell_commands(self):
+    def as_shell_commands(self) -> List[str]:
         return [a.as_shell_assignment() for a in self.assignments]
 
-    def __repr__(self):
+    def __repr__(self) -> str:
```

Comment:
I do not think that typing `__repr__` this is needed

Reply:
It probably gave me warnings if I didn't do it. But I'll try.

Reply:
`cibuildwheel/environment.py:56: error: Function is missing a return type annotation`

Reply:
I noticed above that some functions were marked as `-> None`, which seemed... too obvious and cluttery. As does this. Maybe we can get a bit more granular with the mypy rules - `--strict` just enables them all

Reply:
`mypy.ini` or `setup.cfg` should allow for that, indeed.

Reply:
It looks like the function types really are required, if we want type inference-

> Type inference is not used in dynamically typed functions (those without a function type annotation) — every local variable type defaults to Any in such functions. Any is discussed later in more detail.

https://mypy.readthedocs.io/en/stable/type_inference_and_annotations.html#type-inference

Reply:
Yuk :-(
Somehow it isn't necessary for '__init__`, I think?